packages/breadcrumbs/src/Breadcrumbs.ts

```diff
@@ -0,0 +1,294 @@
+    }
+
+    /**
+     * Always display the first breadcrumb
+     */
+    @property({ type: Boolean, attribute: 'show-root' })
+    public showRoot = false;
+
+    /**
+     * Override the maximum number of visible items
+     */
+    @property({ type: Number, attribute: 'max-visible-items' })
+    public maxVisibleItems = MAX_VISIBLE_ITEMS;
+
+    /**
+     * Accessible name for the Breadcrumbs component
+     */
+    @property({ type: String })
+    public label = '';
+
+    /**
+     * Change the default label of the action menu
+     */
+    @property({ type: String, attribute: 'menu-label' })
+    public menuLabel = 'More items';
+
+    @queryAssignedElements({ selector: 'sp-breadcrumb-item' })
+    private breadcrumbsElements!: BreadcrumbElement[];
+
+    @queryAssignedElements({ slot: 'root', selector: 'sp-breadcrumb-item' })
+    private rootElement!: BreadcrumbElement[];
+
+    @query('#list')
+    private list!: HTMLUListElement;
+
+    @state()
+    private items: BreadcrumbItem[] = [];
+
+    @state()
+    private visibleItems = 0;
+
+    private resizeObserver: ResizeObserver | undefined;
+    private firstRender = true;
+
+    private menuRef: Ref<ActionMenu> = createRef();
+
+    private get hasMenu(): boolean {
+        return this.visibleItems < this.breadcrumbsElements.length;
+    }
+
+    override connectedCallback(): void {
+        super.connectedCallback();
+
+        if (!this.hasAttribute('role')) {
+            this.setAttribute('role', 'navigation');
+        }
+
+        this.updateComplete.then(() => {
+            if (this.showRoot) {
+                this.breadcrumbsElements[0].setAttribute('slot', 'root');
+            }
+        });
+
+        this.resizeObserver = new ResizeObserver(() => {
+            if (this.firstRender) {
+                // Don't adjust overflow on first render, it is adjused in slotChangeHandler
+                this.firstRender = false;
+                return;
+            }
+            this.adjustOverflow();
+        });
+
+        this.resizeObserver.observe(this);
+    }
+
+    public override disconnectedCallback(): void {
+        this.resizeObserver?.unobserve(this);
+        super.disconnectedCallback();
+    }
+
+    override updated(changes: PropertyValues): void {
+        super.updated(changes);
+
+        if (changes.has('label')) {
+            this.setAttribute('aria-label', this.label || 'Breadcrumbs');
+        }
+
+        // Breadcrumbs items were added / removed, or available space changed
+        if (changes.has('visibleItems') || changes.has('items')) {
+            this.items.forEach((item, index) => {
+                this.breadcrumbsElements[index].isLastOfType =
+                    index === this.breadcrumbsElements.length - 1;
+
+                if (!item.isVisible) {
+                    this.breadcrumbsElements[index].setAttribute('hidden', '');
+                } else {
+                    this.breadcrumbsElements[index].removeAttribute('hidden');
```

Comment:
Open to any other suggestions here.

Reply:
would [toggleAttribute()](https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute) work here?